### DISCUSSION TITLE
Process exits without reading anything from chat.

### DISCUSSION THREAD

**Discussion by XcomReborn** [Bukk94/KickLib#4] [Q&A] [answered]
I am trying to connect to a chat room and simply read the messages using the example code.

```
using KickLib.Interfaces;
using KickLib;
using KickLib.Client.Interfaces;
using KickLib.Client.Models.Args;
using KickLib.Client;

Console.WriteLine("Starting!");

IKickApi kickApi = new KickApi();

var userName = "xereborn";

var channelInfo = await kickApi.Channels.GetChannelInfoAsync(userName);

Console.WriteLine (channelInfo.Chatroom.Id);

IKickClient client = new KickClient();

client.OnMessage += delegate (object sender, ChatMessageEventArgs e)
{
    Console.WriteLine(e.Data.Content);
};

await client.ListenToChatRoomAsync(channelInfo.Chatroom.Id);

await client.ConnectAsync();

Console.WriteLine("Exiting...");
```

However this just exits immediately, am I doing something wrong?

**Comment by Bukk94** [comment, marked as answer]
Your example seems correct @XcomReborn. The reason why it exits immediately is that listening is non-blocking. You need to run this code as background task and keep your console up and running, or add some blocking calls to avoid application exit.

Try adding infinite loop after `await client.ConnectAsync()` line. Something like:
```csharp
while(true) 
{
   await Task.Delay(500);
}
```
